### PR TITLE
(feat) exclude namespaced tags from "+N more" cap on post cards (#168)

### LINKED ISSUE #168: Exclude namespaced tags from "3 visible + '+N more'" cap on post cards (REQ-CONTENT-MODEL-03)
## Context

PDR-009 § 7 requires that namespaced tags (`chapter:*`, `wwh:*`) render as distinct badges (via W-H) and are EXCLUDED from the existing REQ-INDEX-05 "3 visible + '+N more'" cap on post cards. Otherwise a post with 2 namespaced tags + 3 free-form tags would show "+3 more" and hide 2 useful free-form tags. Launch-gating.

**Source**: PDR-009 § Implementation Pending item 11 + § 7 + PRD REQ-CONTENT-MODEL-03 (see H-F).

## Work

In post card rendering (cross-cuts with W-H):

1. Split tags into namespaced (`{ns}:{slug}` where `{ns}` is reserved per taxonomy) vs free-form at render time.
2. Namespaced tags render as dedicated badges (W-H scope).
3. Free-form tags feed the existing "3 visible + '+N more'" count; namespaced tags do NOT.
4. Edge case: post with only namespaced tags + no free-form → no "+N more" indicator (0 hidden).

## Acceptance criteria

- [ ] Given a post with 3 free-form + 2 namespaced tags: card shows all 3 free-form (no "+N more") + 2 badges
- [ ] Given a post with 5 free-form + 1 namespaced tag: card shows 3 free-form + "+2 more" + 1 badge
- [ ] Given a post with only 2 namespaced tags: card shows 2 badges, no "+N more"
- [ ] Matches REQ-CONTENT-MODEL-03 in PRD (H-F)
- [ ] No layout regression in existing post cards

## Launch-gating

**Yes — LAUNCH-GATING** per PDR § 7. First post shipping without this produces either hidden-useful-free-form-tags regression or incorrect counts.

### PR DESCRIPTION
**Pull request by alexey-pelykh**
Closes #168.

## Summary

`PostCard.astro` / `PostCardFeatured.astro` now split incoming `tags` via the new `partitionTags(tags)` helper (`src/lib/taxonomy.ts`) and pass only the **free-form** group to `<TagList>`. The existing `MAX_VISIBLE = 3` cap on `TagList` therefore counts free-form tags only — namespaced tags (`chapter:*`, `wwh:*`) are kept out of the cap so they cannot push useful free-form tags behind a "+N more" indicator.

Per **PDR-009 § 7 / REQ-CONTENT-MODEL-03** (HQ PRD H-F; private repo, see `CONTRIBUTING.md` § Cross-repo setup).

## Files

| Path | Change |
| --- | --- |
| `src/lib/taxonomy.ts` | New exports: `RESERVED_NAMESPACES`, `isNamespacedTag(tag)`, `partitionTags(tags)`. JSDoc cites PDR-009 § 7 and the cap-exclusion contract. |
| `src/lib/taxonomy.test.ts` | NEW — 19 vitest unit tests: edge cases for `isNamespacedTag` (empty ns, empty slug, case-sensitivity, multi-colon, unreserved-ns) + order/empty/mixed-input for `partitionTags` + three explicit AC1/AC2/AC3 scenarios. |
| `src/components/PostCard.astro` | +1 import, +4-line splitting block, `tags` → `freeFormTags` in the `<TagList>` prop. |
| `src/components/PostCardFeatured.astro` | Same pattern as `PostCard.astro`. |

## AC verification

| # | AC | Verdict | Evidence |
| --- | --- | --- | --- |
| 1 | 3 free-form + 2 namespaced ⇒ 3 free-form (no "+N more") + 2 badges | **CODE-PASS / VISUAL-PARTIAL** | `taxonomy.test.ts` AC1: `freeForm.length === 3`, `namespaced.length === 2`. `PostCard.astro` passes only `freeFormTags` ⇒ `TagList` renders 3 chips and `remaining = 0` ⇒ no "+N more" element. **Badge rendering** for the namespaced 2 = #166 (W-H scope) — see § Scope split below. |
| 2 | 5 free-form + 1 namespaced ⇒ 3 free-form + "+2 more" + 1 badge | **CODE-PASS / VISUAL-PARTIAL** | AC2 test: `freeForm.length === 5`, `namespaced.length === 1`. `TagList` math: 5 - 3 = 2 ⇒ "+2 more" rendered. Badge = #166. |
| 3 | Only 2 namespaced tags ⇒ 2 badges, no "+N more" | **CODE-PASS / VISUAL-PARTIAL** | AC3 test: `freeForm.length === 0`, `namespaced.length === 2`. `freeFormTags.length > 0` guard correctly skips `<TagList>` rendering when only namespaced exist. Badges = #166. |
| 4 | Matches REQ-CONTENT-MODEL-03 in PRD (H-F) | **PASS** | Inline JSDoc on helpers + comments on cards cite **PDR-009 § 7 / REQ-CONTENT-MODEL-03**. Public statement of the cap-exclusion requirement (issue body) is satisfied. |
| 5 | No layout regression in existing post cards | **PASS** | Diff is surgical (1 import + 4-line computed block + variable rename in `<TagList>` prop on each card). No styling/markup changes. No current post in `src/content/blog/` uses namespaced tags ⇒ live build is byte-unchanged. Local `npm run test` (183/183), `npm run typecheck` (0 errors), `npm run build` (5 pages OK), `npx prettier --check` clean. CI gates Visual (QA-09) / a11y (QA-08) / touch-targets (QA-07) / Lighthouse (QA-06) / audit (QA-10) will run on this PR. |

## Scope split (transparency)

Per the issue body: _"Namespaced tags render as dedicated badges (W-H scope)."_ This PR delivers the **splitting + cap-exclusion semantic only**; **badge rendering** for the namespaced group is **#166 (W-H)** and currently OPEN.

Until #166 lands, namespaced tags do not render visually on cards — strictly better than the prior state, where `chapter:first-moves` would have appeared as a raw chip and counted against the +N cap. After #166 ships, namespaced badges slot into `.post-card-badges` next to `PillarBadge` / `SeriesBadge`; no further changes to PostCard / PostCardFeatured are needed.

The AC#1-3 wording "+ N badges" therefore describes the integrated end state of #166 + #168. AC#1-3 are **CODE-PASS / VISUAL-PARTIAL** here and become **fully VISUAL-PASS** once #166 merges.

## Notes for orchestrator / Wave 2

- Merge strategy: **rebase-merge** (per `CONTRIBUTING.md`).
- No HQ private files are referenced by content (only by docstring citations to PDR-009 § 7 and REQ-CONTENT-MODEL-03).
- No visual baselines need updating; no audit invariants touch tag-list internals (only `.post-card-summary` text identity, which is unaffected).
- Sister tickets in the PDR-009 schema-evolution slate: #164 ✅ (taxonomy.ts SSoT, prerequisite), #165 (W-G — Zod superRefine on `chapter:*`/`wwh:*` slug values), #166 (W-H — labeled badge components), #167 (W-I — BlogFilter chip label resolution), #169 (W-K — JSON-LD), #170 (W-L — llms.txt grouping). This PR does not block any of them; #166 / #167 will consume `RESERVED_NAMESPACES` and `partitionTags` from this PR's exports.

## Test plan

- [x] `npm run test` locally — 183/183 pass (8 files, 19 new in `taxonomy.test.ts`)
- [x] `npm run typecheck` locally — 0 errors, 0 warnings (2 pre-existing unrelated hints in `eslint.config.js` and `BlogFilter.astro`)
- [x] `npm run build` locally — 5 pages built clean
- [x] `npm run lint` locally — 0 errors, 0 warnings
- [x] `npx prettier --check` on touched files — clean
- [ ] CI: full pipeline (security audit, lint, typecheck, unit tests, build, Playwright QA-07/08/09, QA-10.1/10.2/10.3, QA-10.5, Lighthouse QA-06)
- [ ] Automated reviewer (Copilot) feedback addressed if surfaced

---

References:
- **PDR-009 § 7** "Display rendering" (HDIAI HQ, private — see `CONTRIBUTING.md` § Cross-repo setup)
- **PDR-009 § Implementation Pending item 11** (display-cap exclusion identified as launch-gating)
- HQ PRD **REQ-CONTENT-MODEL-03** (display rule for namespaced tags on post cards)
- Sister: #164 (prerequisite, merged), #166 (W-H badge rendering, OPEN)

### CODE REVIEW

File: src/lib/taxonomy.test.ts [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+    expect(partitionTags(['foo:bar', 'pillar:practice'])).toEqual({
+      namespaced: [],
+      freeForm: ['foo:bar', 'pillar:practice'],
+    });
+  });
+
+  describe('REQ-CONTENT-MODEL-03 cap-exclusion scenarios (#168)', () => {
+    // These three cases mirror the post-card AC: the freeForm count drives
+    // the existing "3 visible + +N more" cap on `TagList`; the namespaced
+    // count drives the dedicated badge surface (W-H scope, #166). Both
```
[the remark this answers is not in the document]
Comment (alexey-pelykh) [reply src/lib/taxonomy.test.ts:128]: ✅ Addressed in 1173d51

Applied the suggested fix: `"+N more"` is now properly quoted in the doc comment, eliminating the duplicated `+`. Fixed both instances in `src/lib/taxonomy.test.ts`:

- **Block comment (L127)**: `"3 visible + +N more"` → `"3 visible + '+N more'"` — matches the canonical form in `src/lib/taxonomy.ts:106`.
- **AC2 test name (L144)**: `+ +2 more` → `+ "+2 more"` (double-quoted inside the single-quoted test name).

Tests pass (19/19 in `taxonomy.test.ts`). Comment/doc-only — no runtime or logic change.

---
🤖 Generated with [Claude Code](https://claude.com/claude-code)
Co-Authored-By: Claude Opus 4.7 (1M context) <noreply@anthropic.com>
